frontend/js/src/settings/notification-settings/NotificationSettings.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from "react";
+import { toast } from "react-toastify";
+import { Helmet } from "react-helmet";
+
+import Switch from "../../components/Switch";
+
+export default function NotificationSettings() {
+  const [notificationsEnabled, setNotificationsEnabled] = React.useState(false);
+  const [digestEnabled, setDigestEnabled] = React.useState(false);
+  const [digestAge, setDigestAge] = React.useState<number>(7);
+  const [initialDigestEnabled, setInitialDigestEnabled] = React.useState(false);
+  const [initialDigestAge, setInitialDigestAge] = React.useState<number | null>(
+    null
+  );
+  const [loading, setLoading] = React.useState(true);
+  const [saving, setSaving] = React.useState(false);
+
+  React.useEffect(() => {
+    async function fetchDigestSettings() {
+      try {
+        const response = await fetch("/settings/digest-setting/");
+        if (!response.ok) {
+          throw new Error(`${response.status} HTTP response.`);
+        }
+        const data = await response.json();
+        setDigestEnabled(data.digest);
+        setDigestAge(data.digest_age);
+        setInitialDigestEnabled(data.digest);
+        setInitialDigestAge(data.digest_age);
+        setNotificationsEnabled(data.digest);
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error("Could not fetch notification settings.", error);
+        toast.error("Failed to load notification settings.");
+      } finally {
+        setLoading(false);
+      }
+    }
+    fetchDigestSettings();
+  }, []);
+
+  const updateDigestSettings = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setSaving(true);
+    try {
+      const response = await fetch("/settings/digest-setting/", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          digest: digestEnabled && notificationsEnabled,
+          digest_age: digestAge,
+        }),
+      });
+      if (!response.ok) {
+        throw new Error(`${response.status} HTTP response.`);
+      }
+      toast.success("Notification settings saved successfully");
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error("Could not update notification settings.", error);
+      toast.error("Failed to save notification settings. Please try again.");
+    } finally {
+      setSaving(false);
+    }
+  };
+
+  const hasNoChanges =
+    digestEnabled === initialDigestEnabled && digestAge === initialDigestAge;
+
+  if (loading) {
+    return <div>Loading notification settings...</div>;
+  }
+
+  return (
+    <>
+      <Helmet>
+        <title>Notification Settings</title>
+      </Helmet>
+      <h2 className="page-title">Notification settings</h2>
```

Comment:
What is this css class from?
```suggestion
      <h2>Notification settings</h2>
```

Reply:
https://github.com/metabrainz/listenbrainz-server/blob/1235c47ed65db9d4ba93fd06f1b0989bb5213af6/frontend/js/src/settings/brainzplayer/BrainzPlayerSettings.tsx#L194

I thought it was a custom defined css class.

Reply:
My bad, sorry! Just saw it is used everywhere, but not defined.

Reply:
No problem!